src/test/java/org/quiltmc/loader/impl/config/ConfigTester.java

```diff
@@ -0,0 +1,245 @@
+/*
+ * Copyright 2016 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.quiltmc.loader.impl.config;
+
+import java.nio.file.Path;
+import java.util.ArrayList;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+import org.quiltmc.loader.api.LoaderValue;
+import org.quiltmc.loader.api.config.Config;
+import org.quiltmc.loader.api.config.ConfigWrapper;
+import org.quiltmc.loader.api.config.Constraint;
+import org.quiltmc.loader.api.config.TrackedValue;
+import org.quiltmc.loader.api.config.annotations.Comment;
+import org.quiltmc.loader.api.config.values.ValueList;
+import org.quiltmc.loader.impl.QuiltLoaderImpl;
+
+public class ConfigTester {
+	@TempDir
+	static Path TEMP;
+
+	static TrackedValue<Integer> TEST_INTEGER;
+	static TrackedValue<Boolean> TEST_BOOLEAN;
+	static TrackedValue<String> TEST_STRING;
+	static TrackedValue<ValueList<Integer>> TEST_LIST;
+
+	public ConfigTester() {
+		QuiltLoaderImpl.INSTANCE.setGameProvider(new DummyGameProvider(TEMP));
+	}
+
+	@Test
+	public void testValidation() {
+		Assertions.assertThrows(RuntimeException.class, () -> {
```

Comment:
I'm not a huge fan of expecting a RuntimeException. This will pass if there's *any* uncaught exception in the test. Would be nice to catch a more specific exception (with assertThrowsExactly as well).